clippy_lints/src/raw_strings.rs

```diff
@@ -71,6 +71,23 @@ impl RawStrings {
 
 impl EarlyLintPass for RawStrings {
     fn check_expr(&mut self, cx: &EarlyContext<'_>, expr: &Expr) {
+        if let ExprKind::FormatArgs(format_args) = &expr.kind
+            && !in_external_macro(cx.sess(), format_args.span)
+            && format_args.span.check_source_text(cx, |src| src.starts_with('r'))
+            && let Some(str) = snippet_opt(cx.sess(), format_args.span)
+        {
+            let count_hash = str.bytes().skip(1).take_while(|b| *b == b'#').count();
+            let str = &str[count_hash + 2..str.len() - count_hash - 1];
```

Comment:
Let's go with a `.get` to guard against any proc macro shenanigans causing a panic